File: q1481/q1481.py
class Solution:
    def findLeastNumOfUniqueInts(self, arr: List[int], k: int) -> int:
        numCount = Counter(arr)
        counts = sorted(list(numCount.values()))
        res = len(counts)
        for i in range(len(counts)):
            # Get the smallest count possible
            count = counts[i]
            # If the count is within the capcity of k, remove 1 element
            if count <= k:
                res -= 1
                k -= count
            # If the count is not within the capacity of k, complete the operation
            else: 
                break
        return res

File: q1481/q1481.java
class Solution {
    public int findLeastNumOfUniqueInts(int[] arr, int k) {
        Map<Integer, Integer> numCount = new HashMap<>();
        for (int num : arr)
            numCount.put(num, numCount.getOrDefault(num, 0) + 1);
        List<Integer> counts = new ArrayList<>();
        counts.addAll(numCount.values());

        // Collections.sort(counts);

        counts.sort((a, b) -> Integer.compare(a, b));

        int res = counts.size();

        // Get the smallest count possible
        for (int count : counts) {
            // If the count is within the capcity of k, remove 1 element
            if (k >= count) {
                res -= 1;
                k -= count;
            }
            // If the count is not within the capacity of k, complete the operation
            else {
                break;
            }
        }
        return res;
    }
}